cmd/crank/beta/render/load.go

```diff
@@ -129,6 +131,24 @@ func LoadYAMLStream(fileOrDir string) ([][]byte, error) {
 	return out, nil
 }
 
+// getYAMLFiles returns a list of YAML files in the supplied directory, ignoring any subdirectory.
+func getYAMLFiles(dir string) (yamls []string, err error) {
+	entries, err := os.ReadDir(dir)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot read directory")
+	}
+	for _, entry := range entries {
+		if entry.IsDir() {
+			// We don't care about nested directories.
+			continue
+		}
+		if ext := strings.ToLower(filepath.Ext(entry.Name())); ext == ".yaml" || ext == ".yml" {
+			yamls = append(yamls, filepath.Join(dir, entry.Name()))
+		}
+	}
```

Comment:
Looks like `filepath.Glob` not working with multiple extensions, we could still leverage it here as follows:

```suggestion
	var yamls []string
	var ext := []string{"yaml", "yml"}

	for _, e := range ext {
		y, err := filepath.Glob(filepath.Join(dir, "*."+e))
		if err != nil {
			return nil, err
		}
		yamls = append(yamls, y...)
	}
	
	return yamls, nil
```

Reply:
I also thought it would have been clearer, but then I realised this would result in all `.yml` files being sorted after all `.yaml` files in case of mixed usage, which depending on the caller's usage of the slice returned, could be undesirable, and didn't make much sense to just sort the files later. `ReadDir` [guarantees](https://golang.org/pkg/os#ReadDir) we iterate over entries in alphabetical order.

Both [ReadDir](https://pkg.go.dev/github.com/spf13/afero#ReadDir) and [Glob](https://pkg.go.dev/github.com/spf13/afero#Glob) have their counterpart in afero, so any of the two approaches would be ok w.r.t. that.

The other option I've considered was going with WalkDir, which would allow to support subdirectories easily, but I wanted to preserve the current "glob-like" behaviour for now.

If you feel strongly about the `filepath.Glob` approach I can switch, LMK.

Reply:
👍  I'll just add a TODO to revisit it when we move to an `afero.FS` and some more context

Reply:
Looks like this is still open - I couldn't see a related comment or Todo (which is no longer a todo though)

Reply:
Yes, you are right, got lost in the refactoring, let me add a comment and then I'll merge this.